### PR TITLE
11206 intent list

### PR DESCRIPTION
**Pull request by VitorLamego**
**Proposed changes**:
-  As described in #11206, now you can describe intent as a list instead of numerous mappings of the same type

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Review by losterloh — CHANGES_REQUESTED** [review]
This is a change to the data format, so `LATEST_TRAINING_DATA_FORMAT_VERSION` defined in `rasa.shared.constants` will need to be increased.
(I am currently myself unclear about the versioning scheme for the data format and will get back to you on this topic once I have clarity.)

Also, because it is a change to the data format, the documentation (probably at docs/docs/domain.mdx) needs to be updated to reflect this change.

File: rasa/shared/core/slot_mappings.py [hunk 1]
```diff
@@ -202,19 +202,32 @@ def check_mapping_validity(
...
+                mapping_type == SlotMappingType.FROM_INTENT
+                and intent not in domain.intents
+                ):
+                    rasa.shared.utils.io.raise_warning(
+                        f"Slot '{slot_name}' uses a 'from_intent' mapping for "
+                        f"a non-existent intent '{mapping.get('intent')}'. "
+                        f"Skipping slot extraction because of invalid mapping."
+                    )
+                    return False
+            return True
```
[the remark this answers is not in the document]
Comment (AlGouvea) [reply rasa/shared/core/slot_mappings.py:230]: Hello, I've been working on final fixes on this issue and have changed the code according to your recommendation, however it's still unclear to me which parts of the domain.mdx files will need to be changed.
Also, as you mentioned about the ```LATEST_TRAINING_DATA_FORMAT_VERSION``` versioning format, have you had any updates?

File: changelog/11206.improvement.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Update mappings of the same type to present intent as a list instead of other mappings.
```
Comment (ancalita) [inline changelog/11206.improvement.md:1]: Currently the phrasing is a bit unclear, I'd rephrase:
```suggestion
Allow slot mapping parameter `intent` to accept a list of intent names (as strings), in addition to accepting an intent name as a single string.
```

File: rasa/shared/core/slot_mappings.py [hunk 3]
```diff
@@ -202,20 +202,23 @@ def check_mapping_validity(
...
-            mapping_type == SlotMappingType.FROM_INTENT
-            and mapping.get(INTENT) not in domain.intents
-        ):
-            rasa.shared.utils.io.raise_warning(
-                f"Slot '{slot_name}' uses a 'from_intent' mapping for "
-                f"a non-existent intent '{mapping.get('intent')}'. "
-                f"Skipping slot extraction because of invalid mapping."
-            )
-            return False
+        if mapping_type is SlotMappingType.FROM_INTENT:
```
Comment (ancalita) [inline rasa/shared/core/slot_mappings.py:205]: This was correct with `==` comparison operator, why the change to `is`?

File: rasa/shared/core/slot_mappings.py [hunk 4]
```diff
@@ -202,20 +202,23 @@ def check_mapping_validity(
...
+            if isinstance(mapping.get(INTENT), list):
+                Mapping = mapping.get(INTENT)
+            else:
+                Mapping = [mapping.get(INTENT)]
+            
+            for intent in Mapping:
```
Comment (ancalita) [inline rasa/shared/core/slot_mappings.py:211]: This can be rewritten as suggested earlier by Lukas:
```suggestion
            intent_names = SlotMapping.to_list(mapping.get(INTENT))
            for intent in intent_names:
```

File: tests/core/test_actions.py [hunk 5]
```diff
@@ -2665,6 +2665,65 @@ async def test_action_extract_slots_non_required_form_slot_with_from_entity_mapp
     )
     assert events == [SlotSet("form1_info1", "info1"), SlotSet("form1_slot1", "Filled")]
 
+async def test_action_multiple_intents_same_values():
```
Comment (ancalita) [inline tests/core/test_actions.py:2668]: This is a good test to have, we also need a unit test in `tests/shared/core/test_slot_mappings.py` that `check_mapping_validity` method gives the correct result in case of valid and invalid inputs.

**Review by ancalita — CHANGES_REQUESTED** [review]
@VitorLamego thank you for the updates, I'll be the main reviewer going forward on this PR.
I've left a few comments on the existing state.

I can also confirm that the `LATEST_TRAINING_DATA_FORMAT_VERSION` constant needs to be updated to `3.2` as part of this PR (this is because of the separate semantic versioning we apply to training data format changes, more info [here](https://rasa.com/docs/rasa/migration-guide/#upgrading-version-from-20-to-30)). 

The change to the parameter `intent` in slot mappings needs to be documented in `docs/docs/domain.mdx`, for every slot mapping type where this parameter is explained, for example L551 and L662. Perhaps a good idea is to add a Markdown info note in the Slot Mappings headline announcing that intent parameter now allows list of intent names with the caption `New in 3.3`.

**Comment by ancalita** [comment]
@VitorLamego Also please note that the `main` branch was force-pushed recently, so you'll need to rebase before making any new review changes.
